src/core/org/apache/jmeter/util/JSR223TestElement.java

```diff
@@ -86,11 +88,16 @@ public JSR223TestElement() {
     }
 
     protected ScriptEngine getScriptEngine() throws ScriptException {
-        final String lang = getScriptLanguage();
+        String lang = getScriptLanguage();
+        
+        if (lang.isEmpty()) {
+            lang = defaultScriptLanguage;
+            log.warn("Script language has not been chosen on the UI, the script will be interpreted as a groovy script");
```

Comment:
>it's not the best solution for performance

JMeter is meant for performance testing. Why do you break the main use case?

1) The `warn` message does not enable user to identify which particular sampler is bad, so it will be impossible to fix large JMeter script. That kind of message is useless.
2) The message should be printed at most once for each sampler for each test execution.